### PR TITLE
Update CI scripts

### PR DESCRIPTION
**Pull request by NelsonVides**

### CODE REVIEW

**Comment by NelsonVides** [comment]
@big-r81 hi there! Related to #4, I've been trying here to modernise CI and now also adding windows to it, so that we can know its build won't be broken by any future changes. However, I cannot get windows to build in this runner, it keeps complaining that it can't find the <openssl/sha.h> header files. Do you know how does your windows build manage to find this header? Perhaps you can help me here :)

**Comment by big-r81** [comment]
Hi, that's great to hear 🫶. We use [vcpkg](https://vcpkg.io/) to install the OpenSSL 3 dev dependency. Afterwards we add the dirs to the [env paths](https://github.com/apache/couchdb-glazier/blob/main/bin/shell.ps1#L37-L39) so that the compiler can use it.

File: rebar.config [hunk 1]
```diff
@@ -12,11 +12,15 @@
...
+      {rebar3_codecov, "0.6.0"}]},
       {port_env,
         [
-          {"CFLAGS", "$CFLAGS -std=c99 -O3 -g -Wall -Wextra -fPIC -I/opt/homebrew/include -I/usr/local/include --coverage"},
-          {"LDLIBS", "$LDLIBS -lcrypto -L/opt/homebrew/lib/ -L/usr/local/lib --coverage"}
+          {"(linux|solaris|freebsd|netbsd|openbsd|dragonfly|darwin|gnu)",
+               "$CFLAGS -std=c99 -O0 -g -Wall -Wextra -fPIC --coverage"},
+          {"(linux|solaris|freebsd|netbsd|openbsd|dragonfly|darwin|gnu)",
+               "$LDLIBS -lcrypto --coverage"},
+          {"win32", "CFLAGS", "$CFLAGS ${OPENSSL_INSTALL_DIR}\include /O0 /DNDEBUG /Wall"},
```
Comment (big-r81) [inline rebar.config:22]: Maybe

```suggestion
          {"win32", "CFLAGS", "$CFLAGS /I${OPENSSL_INSTALL_DIR}\include /O2 /DNDEBUG /Wall"},
```

Comment (big-r81) [inline rebar.config]: @NelsonVides 
this should work:

```
{"win32", "CFLAGS", "$CFLAGS /I${OPENSSL_INSTALL_DIR}/include /O2 /DNDEBUG /Wall"}
```

**Comment by big-r81** [comment]
or with special escaping:

```
{"win32", "CFLAGS", "$CFLAGS /I\"${OPENSSL_INSTALL_DIR}\\include\" /O2 /DNDEBUG /Wall"}
```

**Comment by big-r81** [comment]
@NelsonVides Where do you set your env var `OPENSSL_INSTALL_DIR`?

**Comment by NelsonVides** [comment]
@big-r81 
I've tried with several mechanisms but it keeps failing with
```
d:/a/fast_pbkdf2/fast_pbkdf2/c_src/fast_pbkdf2.c(35): fatal error C1083: Cannot open include file: 'openssl/sha.h': No such file or directory
```
I was expecting the env-vars to be set by any of the many gha plugins started here, or when installed by choco or vspkg, but none works. I haven't touched a windows machine in over 5 years so I honestly don't know much about how windows work here 🥲

**Comment by big-r81** [comment]
@NelsonVides 
For example, if you install OpenSSL with `choco` you need to check where it is installed. Then you could set the path to the env var `OPENSSL_INSTALL_DIR`.

**Comment by big-r81** [comment]
@NelsonVides In #6 there is a version which works for the Windows runner. You can use this as an idea. :wink:

**Comment by NelsonVides** [comment]
@big-r81 Thanks for the PR! I cherry-picked your commit and rebased on top of the main branch, you're tagged there as a co-author in the commit :)
CI stopped running here, GHA didn't like my rebasing, so I created a new PR at #7

**Pull request closed without merging**